### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by johnantonn**
Let's merge these changes in the API:
- Additional eeris route required for the admin dashboard web app.
- Flag property in a project to indicate the need for historical averages calculation (generalization of eeris-only functionality).
- Drop table code/script for the permanent deletion of data from cockroachdb.

### CODE REVIEW

File: server/routes/events.js [hunk 1]
```diff
@@ -132,6 +132,7 @@ router.post("/:EVENT_COLLECTION", (req, res) => Project.findOne({ projectId: req
     }
     cenote.id = uuid();
     payload[i].cenote = { ...cenote };
+    if (project.redisHist) payload[i].redisHist = true;
```
Comment (iamnapo) [inline server/routes/events.js:135]: `redisHist` is an implementation detail, it's not part of the payload. As such, I think you should put it inside the `cenote` object.

File: server/routes/queries.js [hunk 2]
```diff
@@ -866,6 +865,72 @@ router.get("/eeris", canAccessForCollection, (req, res) => Project.findOne({ pro
...
+      const jsonValue = JSON.parse(value);
+      finalDate.setDate(finalDate.getDate() + 1);
+      while (date.getTime() !== finalDate.getTime()) {
+        const year = date.getFullYear();
+        const month = (`0${date.getMonth() + 1}`).slice(-2);
+        const day = (`0${date.getDate()}`).slice(-2);
+        const avg = jsonValue[`avg_${year}-${month}-${day}`] || 0;
+        values.push(avg);
+        date.setDate(date.getDate() + 1);
+      }
```
Comment (iamnapo) [inline server/routes/queries.js:925]: `getTime` returns the UNIX timestamp in milliseconds, but in each iteration you are only changing the day of the month. Won't this lead to an infinite loop?

File: server/routes/queries.js [hunk 3]
```diff
@@ -933,7 +998,7 @@ router.delete("/testCleanup", async (req, res) => {
       await r.del(redisKeyeeRIS);
     }
     const query = `DROP TABLE IF EXISTS ${req.params.PROJECT_ID}_${eventCollection}`;
-    await client.query(query);
+    client.query(query);
```
Comment (iamnapo) [inline server/routes/queries.js:1001]: If this (`DELETE /testCleanup`) is a slow request that you don't want to wait for a response, then you should handle that client-side, not server-side.

**Review by iamnapo — CHANGES_REQUESTED** [review]

File: server/routes/queries.js (hunk 2 above)
Comment (johnantonn) [reply server/routes/queries.js:925]: The change of day will propagate to a change in month and year accordingly (see screenshot below). However, the comparison operator "!==" should be replaced by "<", I think it's safer.

date

File: server/routes/events.js (hunk 1 above)
Comment (johnantonn) [reply server/routes/events.js:135]: Ok, the historical averages functionality will be put under the microscope in its entirety, so I'll take care of it.

File: server/routes/queries.js (hunk 2 above)
Comment (iamnapo) [reply server/routes/queries.js:925]: It does, but `getTime()` also takes into account the time:

```js
new Date("2020-11-08T06:47:02.609Z").getTime() !== new Date("2020-11-08T05:00:01.406Z").getTime()
```

However, I just noticed that this endpoint accepts dates as `YYYY-MM_DD` and not as `ISOString`, like every other (in the `timeframe` parameter). In that case the time will default to `00:00:00.000Z`, so thats why this happens to work correctly.

File: server/routes/queries.js (hunk 3 above)
Comment (johnantonn) [reply server/routes/queries.js:989]: Will examine later as part of testing..

File: server/routes/events.js (hunk 1 above)
Comment (johnantonn) [reply server/routes/events.js]: Done.

**Comment by iamnapo** [comment]
@johnantonn These `ERESOLVE` come from Travis upgrading to node v15. Downgrade it to lts, revert the last 3 commits and it should be gone. You can do:

1. `git revert --no-commit HEAD~3..` To revert changes from last 3 commits.
2. Change `node` to `lts/*` here: https://github.com/AuthEceSoftEng/cenote-api/blob/dev/.travis.yml#L5.
3. Commit the new changes.

**Pull request merged**